src/schema/v2/partner.ts

```diff
@@ -150,6 +165,24 @@ export const PartnerType = new GraphQLObjectType<any, ResolverContext>({
             gravityArgs.artwork_id = flatten([args.artworkIDs])
           }
 
+          if (args.shallow === false && partnerArtworksAllLoader) {
+            return partnerArtworksLoader(id, gravityArgs).then(({ body }) => {
+              const artworkIds = body.map((artwork) => artwork._id)
+              const gravityArtworkArgs = {
+                artwork_id: artworkIds,
+              }
+
+              return partnerArtworksAllLoader(id, gravityArtworkArgs).then(
+                ({ body }) => {
+                  return connectionFromArraySlice(body, args, {
+                    arrayLength: artworkIds.length,
+                    sliceStart: 0,
+                  })
```

Comment:
Any issues with leaving `arrayLength` and `sliceStart` as these are?

Reply:
Does it work if you use the same `offset` and `parseInt(headers["x-total-count"] || "0", 10)` from the response below? That way clients will still be able to paginate and use basic connection features.

(Make sure that we're referencing the `headers` from the promise response from `partnerArtworksLoader` and not from `partnerArtworksAllLoader`. I think this is why it was broken on our first attempt)

Reply:
Thank you Sarah, using headers from the correct promise works like a charm :]